service/geopmdpy_test/geopmdpy_test.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#  Copyright (c) 2015 - 2021, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+test_name=$(basename $0)
+test_dir=$(dirname $0)
+obj_dir=$(readlink -f $test_dir/../..)
+lib_path=$obj_dir/.libs
+real_path=$(dirname $(readlink -f $0))
+top_dir=$(readlink -f $real_path/../..)
+
+test_class=$(echo $test_name | awk -F. '{print $1}')
+service_dir=$top_dir/service
+lib_dir=$obj_dir/.libs
+export PYTHONPATH=$service_dir:$PYTHONPATH
+export LD_LIBRARY_PATH=$lib_dir:$LD_LIBRARY_PATH
+
+# PYTHON is expected to be set when running make check. However, the default
+# value may be used when invoking this script directly from the command line.
+"${PYTHON-python3}" $service_dir/geopmdpy_test/$test_class.py --verbose $test_name >& $test_dir/$test_name.log
```

Comment:
```suggestion
"${PYTHON-python3}" ${service_dir}/geopmdpy_test/${test_class}.py --verbose ${test_name} >& ${test_dir}/${test_name}.log
```

Adding ```{}``` where appripriate.  This should be done everywhere.  I'm not a fan of expansions like: ```$test_class.py``` where ```test_class``` is the BASH variable.  This can/will lead to subtle expansion problems later and should be taken care of now.